manager/manager.go

```diff
@@ -89,7 +89,12 @@ func (m *Manager) createPods(ctx context.Context) error {
 	podLabels := map[string]string{
 		"app": m.podApplication,
 	}
-
+	// The "app" label might be overwritten by the pod template
+	if podTemplate.Template.Labels != nil {
+		for k, v := range podTemplate.Template.Labels {
+			podLabels[k] = v
+		}
```

Comment:
I think we should set the `"app": m.podApplication,` after this... manager relies on the app label to find these pods... we shouldn't allow bad podTemplates to cause infinite pod creation

Reply:
Fixed